webapp/src/modules/order/sagas.spec.ts

```diff
@@ -0,0 +1,206 @@
+
+    it("should put the execute order failure with an error message saying that the order doesn't match the NFT", () => {
+      return expectSaga(orderSaga)
+        .put(
+          executeOrderFailure(order, nft, 'The order does not match the NFT')
+        )
+        .dispatch(executeOrderRequest(order, nft, fingerprint))
+        .run({ silenceTimeout: true })
+    })
+  })
+
+  describe("and the nft doesn't have the same token id as the order", () => {
+    beforeEach(() => {
+      order.tokenId = 'anotherTokenId'
+    })
+
+    it("should put the execute order failure with the an error message saying that the order doesn't match the NFT", () => {
+      return expectSaga(orderSaga)
+        .put(
+          executeOrderFailure(order, nft, 'The order does not match the NFT')
+        )
+        .dispatch(executeOrderRequest(order, nft, fingerprint))
+        .run({ silenceTimeout: true })
+    })
+  })
+
+  describe('and the order execution fails', () => {
+    let vendor: Vendor<VendorName>
+    let error: Error & { code?: ErrorCode }
+    let errorMessage: string
+
+    beforeEach(() => {
+      vendor = VendorFactory.build(nft.vendor)
+      errorMessage = 'The execution was reverted'
+      error = new Error(errorMessage)
+      error.code = ErrorCode.SALE_PRICE_TOO_LOW
+    })
+
+    it('should put the execute order failure with the error message and the error code', () => {
+      return expectSaga(orderSaga)
+        .provide([
+          [select(getWallet), wallet],
+          [call([VendorFactory, 'build'], nft.vendor), vendor],
+          [
+            call(
+              [vendor.orderService, 'execute'],
+              wallet,
+              nft,
+              order,
+              fingerprint
+            ),
+            throwError(error)
+          ]
+        ])
+        .put(
+          executeOrderFailure(
+            order,
+            nft,
+            errorMessage,
+            ErrorCode.SALE_PRICE_TOO_LOW
+          )
+        )
+        .dispatch(executeOrderRequest(order, nft, fingerprint))
+        .run({ silenceTimeout: true })
+    })
+  })
+
+  describe('and the order execution is successful', () => {
+    let txHash: string
+    let vendor: Vendor<VendorName>
+    let rentalListing: RentalListing
+
+    beforeEach(() => {
+      txHash = 'aTxHash'
+      vendor = VendorFactory.build(nft.vendor)
+    })
+
+    describe('and the NFT has a rental', () => {
+      beforeEach(() => {
+        nft.openRentalId = 'anOpenRentalId'
+        rentalListing = {
+          id: 'anOpenRentalId',
+          status: RentalStatus.CANCELLED
+        } as RentalListing
+      })
+
+      it('should wait for the rental to change its status to cancelled and put the success action', () => {
+        return expectSaga(orderSaga)
+          .provide([
+            [select(getWallet), wallet],
+            [call([VendorFactory, 'build'], nft.vendor), vendor],
+            [select(getRentalById, nft.activeOrderId!), rentalListing],
```

Comment:
I think you intented to put these here:
```suggestion
            [select(getRentalById, nft.openRentalId!), rentalListing],
```